developer_guides/linux_driver/third_party/keyword_detect/keyword_detect.rst

```diff
@@ -0,0 +1,148 @@
+.. _keyword_detect:
+
+Keyword Detection Driver implementation and user guide
+######################################################
+
+.. contents::
+   :local:
+   :depth: 1
+
+Keyword Detection
+*****************
+
+Keyword Detection (KD), also known as Voice Activation or Sound Trigger, is a feature that triggers a speech recognition engine when a predefined keyphrase (keyword) is successfully detected. Offloading the keyphrase detection algorithm to the embedded processing environment (i.e. dedicated DSP) reduces system power consumption while listening for an utterance.
+
+With respect to how to integrate a 3rd party detection algorithm into the SOF firmware, please refer to https://thesofproject.github.io/latest/developer_guides/firmware/kd_integration/index.html 
+
+Keyword Detection pipelines
+***************************
+
+.. code-block:: none
+
+   # PCM  <-----(pipe 8)--------- host <---------------- KPB <------- Volume <--- DMIC (dmic16k)
+   #  |                                                   |
+   #  |                                                   |
+   #  |                                                   |
+   # Detector Sink <---Detector(pipe 9) <--- selector <---+
+
+We are using DAPM events to trigger the detect pipeline(pipe 9). Here the ‘Detector Sink’ is a virtual DAPM widget(visible to the driver, but not to the FW), it is used to send pipeline control IPCs(hw_params, trigger start/stop, hw_free) to firmware. These control IPCs are sent to FW in the sequence like this (1->2(2.1->2.2)->3->4->5->6(6.1->6.2)):
+
+.. csv-table:: DAPM events & stream control sequence
+   :header: "IPCs", "Pipe 8", "Detector Sink event", "Pipe 9"
+   :widths: 20, 10, 25, 10
+
+   "hw_params", "1", "2 (DAPM_PRE_PMU)", "2.1"
+   "trigger start", "3", "", "2.2"
+   "trigger stop", "4", "6 (DAPM_POST_PMD)", "6.1"
+   "hw_free", "5", "", "6.2"
+
+Kcontrols for Keyword Detection
+*******************************
+
+The KWD detection topology contains several kcontrols mainly belonging to the following types:
+
+PGA kcontrol
+============
+
+These are associated with the volume components and are used to adjust the volume after the samples are captured by DMIC.
+
+.. code-block:: none
+
+   numid=12,iface=MIXER,name='PGA8.0 8 KWD Capture Volume'
+
+Bytes kcontrols
+===============
+
+KPB, Selector and Detector are all treated as processing type components in the SOF driver. Each of these components have an associated byte type kcontrol and are configured using the default values from topology as follows:
+
+KPB kcontrol
+------------
+
+The kcontrol for KPB configuration is(amixer controls | grep “KPB”):
+
+.. code-block:: none
+
+   numid=13,iface=MIXER,name='KPBM8.0 KPB'
+
+The initial value of it is in KPB_priv section from sof/tools/topology/sof/pipe-kfbm-capture.m4(The first 32 Bytes are abi header), and it is aligned with the definition of struct sof_kpb_config in sof/include/user/kpb.h.
```

Comment:
Small nits: add a space between a letter and an open parenthesis. here ...``m4 (the first 32 bytes). Please use lower letter for 'The' and 'Bytes'). Same issue below.

Reply:
Thanks.